### PR TITLE
Correct naming conventions in rna-transcription exercise

### LINKED ISSUE #986: Bad naming convention in the nucleotide exercises
Acronyms and Initialisms types should all be UpperCamelCase.  So `DNA` and `RNA` structs should be `Dna` and `Rna`.

Here's the proof: <https://rust-lang.github.io/rfcs/0430-finalizing-naming-conventions.html>

### PR DESCRIPTION
**Pull request by benreyn**
Closes #986

### CODE REVIEW

File: exercises/rna-transcription/example.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/// The possible nucleotides in DNA and RNA
+/// The possible nucleotides in Dna and Rna
```
Comment (coriolinus) [inline exercises/rna-transcription/example.rs:1]: Comments are exceptions; normal acronym rules (all-caps) should apply there.

**Review by coriolinus — COMMENTED** [review]
Please restore comment capitalization. I marked the first instance. Other than that, looks good!

File: exercises/rna-transcription/example.rs (hunk 1 above)
Comment (benreyn) [reply exercises/rna-transcription/example.rs]: Addressed in 64b1b68. Thanks for the review!

**Review by coriolinus — APPROVED** [review]
Looks great, thanks for doing this work!

**Pull request merged**